### PR TITLE
Feature/advanced controlls

### PR DESCRIPTION
**Pull request by mplieske**
added:
- close report an settings on esc
- slider in change with time gets focus on key press
- slider in change with time changes value with step size 5 if ctrl is pressed

### CODE REVIEW

File: src/main/java/de/doubleslash/keeptime/view/ViewController.java [hunk 1]
```diff
@@ -593,6 +623,31 @@ private Node addProjectToProjectList(final Project p) {
 
          dialog.getDialogPane().setContent(vBox);
 
+         slider.setOnKeyPressed(ke -> {
+            if (!slider.isFocused()) {
```
[the remark this answers is not in the document]
Comment (mplieske) [reply src/main/java/de/doubleslash/keeptime/view/ViewController.java]: You are right. The real logic is at line 582 (okButton.setOnKeyPressed(...))

File: src/main/java/de/doubleslash/keeptime/view/ReportController.java [hunk 2]
```diff
@@ -103,9 +108,11 @@ private void initialize() {
       colorTimeLine = new ColorTimeLine(colorTimeLineCanvas);
    }
 
-   private void updateReport(final LocalDate newvalue) {
-      this.currentDayLabel.setText(DateFormatter.toDayDateString(newvalue));
-      final List<Work> currentWorkItems = model.getWorkRepository().findByCreationDate(newvalue);
+   private void updateReport(final LocalDate dateToShow) {
```
[the remark this answers is not in the document]
Comment (mplieske) [reply src/main/java/de/doubleslash/keeptime/view/ReportController.java]: yes indeed it is closeable on esc without the listener 😄.

File: src/main/java/de/doubleslash/keeptime/view/ViewController.java [hunk 3]
```diff
@@ -442,27 +445,47 @@ private void loadSubStages() {
...
+               LOG.info("pressed ESCAPE");
+               reportStage.close();
+            }
+         });
+
+         reportStage.setScene(reportScene);
          reportStage.setTitle("Report");
          reportStage.setResizable(false);
-         reportStage.setOnHiding(e -> this.mainStage.setAlwaysOnTop(true));
+         reportStage.setOnHiding(we -> this.mainStage.setAlwaysOnTop(true));
```
[the remark this answers is not in the document]
Comment (mplieske) [reply src/main/java/de/doubleslash/keeptime/view/ViewController.java]: we is a WindowEvent

File: src/main/java/de/doubleslash/keeptime/view/ViewController.java [hunk 4]
```diff
@@ -188,6 +189,8 @@ public void changeProject(final Project newProject, final long minusSeconds) {
 
    private Map<Project, Node> projectSelectionNodeMap;
 
+   private boolean ctrlIsPressed = false;
```
[the remark this answers is not in the document]
Comment (mplieske) [reply src/main/java/de/doubleslash/keeptime/view/ViewController.java]: DONE! 😄

File: src/main/java/de/doubleslash/keeptime/view/ViewController.java [hunk 5]
```diff
@@ -551,6 +578,9 @@ private Node addProjectToProjectList(final Project p) {
          int gridRow = 0;
          grid.add(new Label("Minutes to transfer"), 0, gridRow);
          final Slider slider = setUpSliderChangeWithTimeMenuItem(activeWorkSecondsProperty);
+         slider.setFocusTraversable(true);
+
+         okButton.setOnKeyPressed(ke -> slider.requestFocus());
```
[the remark this answers is not in the document]
Comment (mplieske) [reply src/main/java/de/doubleslash/keeptime/view/ViewController.java]: the dialog sets the default button on focus even if I say setOnShown(slider.requestFocus()) or setOnSowing(slider.requestFocus()) thats why I did okButton.setOnKeyPressed(...)

File: src/main/java/de/doubleslash/keeptime/view/ViewController.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Death111) [reply src/main/java/de/doubleslash/keeptime/view/ViewController.java]: was still here. i removed it

**Pull request merged**